examples/pytorch/graphsage/dist/train_dist.py

```diff
@@ -284,13 +234,17 @@ def main(args):
         g.rank(), len(train_nid), len(np.intersect1d(train_nid.numpy(), local_nid)),
         len(val_nid), len(np.intersect1d(val_nid.numpy(), local_nid)),
         len(test_nid), len(np.intersect1d(test_nid.numpy(), local_nid))))
+    del local_nid
     if args.num_gpus == -1:
         device = th.device('cpu')
     else:
         dev_id = g.rank() % args.num_gpus
         device = th.device('cuda:'+str(dev_id))
-    labels = g.ndata['labels'][np.arange(g.number_of_nodes())]
-    n_classes = len(th.unique(labels[th.logical_not(th.isnan(labels))]))
+    n_classes = args.n_classes
```

Comment:
Is `n_classes` here different from `n_classes` by unpacking `data` at L125?

Reply:
they're the same. here's in main and obtain `n_classes`, then this value is passed into data. Data is further unpacked in `run()`.